crates/consensus/src/hardforks/isthmus.rs

```diff
@@ -0,0 +1,78 @@
+//! Module containing a [Transaction] builder for the Isthmus network upgrade transactions.
+//!
+//! Isthmus network upgrade transactions are defined in the [OP Stack Specs][specs].
+//!
+//! [specs]: https://specs.optimism.io/protocol/isthmus/derivation.html#network-upgrade-automation-transactions
+//! [Transaction]: alloy_consensus::Transaction
+
+use crate::UpgradeDepositSource;
+use alloc::{string::String, vec::Vec};
+use alloy_eips::eip2718::Encodable2718;
+use alloy_primitives::{address, Address, Bytes, TxKind, B256, U256};
+
+use crate::{Hardfork, TxDeposit};
+
+/// The Isthmus network upgrade transactions.
+#[derive(Debug, Default, Clone, Copy)]
+pub struct Isthmus;
+
+impl Isthmus {
+    /// EIP-2935 From Address
+    pub const EIP2935_FROM: Address = address!("E9f0662359Bb2c8111840eFFD73B9AFA77CbDE10");
+
+    /// Returns the source hash for the Isthmus Deposit Contract deployment.
+    pub fn deposit_contract_source() -> B256 {
+        UpgradeDepositSource { intent: String::from("Isthmus: deposit contract deployment") }
+            .source_hash()
+    }
+
+    /// Returns the EIP-2935 creation data.
+    pub fn eip2935_creation_data() -> Bytes {
+        include_bytes!("./bytecode/eip2935_isthmus.hex").into()
+    }
+
+    /// Returns the list of [TxDeposit]s for the network upgrade.
+    pub fn deposits() -> impl Iterator<Item = TxDeposit> {
+        ([TxDeposit {
```

Comment:
Not needed in this PR, but we will have another upgrade tx for the `L1Block` if the operator fee is included.

Reply:
That's fine and should be a follow-on pr

Reply:
agree, just info - can resolve this one